src/mlpack/methods/ann/layer/transposed_convolution.hpp

```diff
@@ -222,18 +214,23 @@ class TransposedConvolution
    * @param input The input to be padded.
    * @param wPad Padding width of the input.
    * @param hPad Padding height of the input.
+   * @param wExtra The number of extra zeros to the right.
+   * @param hExtra The number of extra zeros to the bottom.
    * @param output The padded output data.
    */
   template<typename eT>
   void Pad(const arma::Mat<eT>& input,
```

Comment:
Maybe we could introduce a `Padding` layer and use that inside the convolution layers to avoid redundant code.
[Pytorch already has support](https://pytorch.org/docs/stable/nn.html#padding-layers) for quite a lot of padding layers. This would also make the implementation more clear.

Reply:
@saksham189 Has there been any discussion on this on irc after you posted this? I think this might be good idea and should be given some thought.

Reply:
Sorry for the slow response, makes totally sense to me.

Reply:
@akhandait I will work on adding the Padding layer and remove the redundant code in a new PR so that it is easy to review.

Reply:
Looks like the Padding layer is done and merged; should we incorporate those changes here, or maybe in another PR after this is merged?  It may be worth opening an issue so we don't forget. :)